lib/OpenQA/WebAPI/Controller/Test.pm

```diff
@@ -388,11 +387,19 @@ sub prepare_job_results {
 
     # prefetch the number of available labels for those jobs
     my $job_labels = $self->_job_labels($jobs);
+    my @job_names = map { $_->TEST } @$jobs;
+
+    # prefetch descriptions from test suites
+    my %desc_args = (name => {in => \@job_names});
+    my @descriptions = $self->db->resultset("TestSuites")->search(\%desc_args, {columns => [qw(name description)]});
+    my %descriptions = map { $_->name => $_->description } @descriptions;
 
     foreach my $job (@$jobs) {
-        my $test   = $job->TEST;
-        my $flavor = $job->FLAVOR || 'sweet';
-        my $arch   = $job->ARCH || 'noarch';
+        my $test    = $job->TEST;
+        my $distri  = $job->DISTRI || 'nodistri';
```

Comment:
same here. DISTRI VERSION FLAVOR ARCH are required fields - and we at least discussed also to include BUILD as required.